File: 30-Days-of-Code/Day-6-Lets-Review/Go/solution.go
package main
import "fmt"

func main() {
    var num uint32
    var firstHalf []byte
    var secondHalf []byte
    
    fmt.Scan(&num)
    strs := make([]string, num)
    
    for i, w := range strs {
        fmt.Scan(&(strs[i]))
        w = strs[i]
        
        for j := 0; j < len(w); j++ {
            if j % 2 == 0 {
                firstHalf = append(firstHalf, w[j])    
            } else {
                secondHalf = append(secondHalf, w[j])
            }
        }
        
        fmt.Printf("%s %s\n", string(firstHalf), string(secondHalf))
        firstHalf = nil
        secondHalf = nil
    }
}

File: 30-Days-of-Code/Day-6-Lets-Review/Python2/solution.py
import sys

n = int(raw_input())

for i in range(1, n+1):
    s = raw_input()
    
    firstHalf = ""
    secondHalf = ""
    
    for j in range(0, len(s)):
        if j % 2 == 0:
            firstHalf += s[j]
        else:
            secondHalf += s[j]
    
    print firstHalf + " " + secondHalf
    
    firstHalf = ""
    secondHalf = ""
